### PR TITLE
qt-6: use makeScopeWithSplicing properly

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

This implements @Artturin's suggestion from 341e6fd558abd.  This commit should cause no eval changes on native builds.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/qt-6/default.nix [hunk 1]
```diff
@@ -174,16 +181,28 @@ let
...
 
-  # TODO(@Artturin): convert to makeScopeWithSplicing'
-  # simple example of how to do that in 5568a4d25ca406809530420996d57e0876ca1a01
-  baseScope = lib.makeScope newScope addPackages;
 
-  bootstrapScope = baseScope.overrideScope'(final: prev: {
+  baseScope = lib.makeScopeWithSplicing' {
+    inherit splicePackages newScope;
+  } {
+    otherSplices = {
```
Comment (Artturin) [inline pkgs/development/libraries/qt-6/default.nix:188]: Check how qt5 does it and omit the unnecessary args

Comment (amjoseph-nixpkgs) [reply pkgs/development/libraries/qt-6/default.nix]: 34d8710  (squashed)

Comment (Artturin) [reply pkgs/development/libraries/qt-6/default.nix]: Remove the unused args and use the prefilled makeScopeWithSplicing' from pkgs also remove the keep and extra

Comment (amjoseph-nixpkgs) [reply pkgs/development/libraries/qt-6/default.nix]: > also remove the keep and extra

I wish I understood what those did.

Comment (amjoseph-nixpkgs) [reply pkgs/development/libraries/qt-6/default.nix]: efdddc051f64

Comment (Artturin) [reply pkgs/development/libraries/qt-6/default.nix]: > > also remove the keep and extra
> 
> I wish I understood what those did.

https://github.com/NixOS/nixpkgs/pull/265045

**Comment by amjoseph-nixpkgs** [comment]
can wait until after branch-off

**Comment by Artturin** [comment]
https://github.com/NixOS/nixpkgs/pull/265524

**Pull request closed without merging**